docs/contents/pdf_model.puml

```diff
@@ -0,0 +1,237 @@
+            **content**: container for elements of type
+            ""Chapter/Table/Figure/Paragraph""
+        }
+
+        class Link {
+            + [int] idx_start
+            + [int] idx_end
+            + [dict] pos_target
+            + [str] libpdf_target
+            --
+            **pos_target**: it's a dictionary with the position info
+            e.g.
+                      page: 3,
+                      x: 300.454
+                      y: 300.454
+
+            **libpdf_target**: points either to an ""Element"" or
+            to a ""Page"". The link is built by concatenating
+            nested elements separated by '/', e.g.
+            ""  chapter.3/chapter.3.2/table.2""
+            For case where the pos_target can not be resolved,
+            the target is set to the target coordinates given as page.<id>/<X>:<Y>
+            ""  page.4/56:789 ""
+        }
+
+        class Paragraph {
+            --
+            **id**: ""paragraph.<1,2,3,n>""
+            A paragraph gets detected by
+            layout analysis. Characters are
+            merged into words, words into
+            lines and lines into paragraphs.
+            A new word, line or paragraph
+            is started if the gap changes
+            significantly.
+        }
+
+        class Chapter {
+            + [str] title
+            + [str] number
+            --
+            **id**: ""chapter.<number>""
+            **number**: globally unique (e.g. 3.2.4)
+            **content**: container for elements of type
+            ""Chapter/Table/Figure/Paragraph""
+        }
+
+        class Cell {
+            + [int] row
+            + [int] col
+            --
+            **row**: 1-based
+            **col**: 1-based
+        }
+
+        class Table {
+            --
+            **id**: ""table.<1,2,3,n>""
+        }
+
+        class Figure {
+            + [str] rel_path
+            + [str] caption
+            --
+            **id**: ""figure.<1,2,3,n>""
+            **rel_path**: figures/<image>
+        }
+        package x <<Layout>>{
+
+             class HorizontalBox {
+            --
+                **text**: plain text in the box
+            }
+
+            class HorizontalLine {
+                --
+                **text**: plain text in the line
+            }
+
+            class Word {
+                --
+                **text**: plain text in the word
+            }
+
+            class Char {
+                --
+                **text**: a plain char
+            }
+        }
+       
+        
+        Char "+chars  1..*" --* "+word  1" Word
+        Word "+words  1..*" --* "+line  1" HorizontalLine
+        HorizontalLine "+line  1..*" --* "+box  1" HorizontalBox
+        
+        HorizontalBox "+box  1" --* "+paragraph  1" Paragraph    
+        HorizontalBox "+box  0..1" --* "+cell  1" Cell
+        HorizontalBox "+box  0..*" -down-* "+figure  1" Figure
+        HorizontalBox "+box  1..*" --* "+chapter  1" Chapter
```

Comment:
How come a chapter can have more than one Box? Were are talking about the title, right?